### PR TITLE
Fix links in docs build

### PR DESCRIPTION
**Pull request by skirby1996**
The docs build needs files to be in a subdirectory of "itwinjs-core" for links to work properly.

### CODE REVIEW

**Comment by skirby1996** [comment]
@mergifyio backport release/3.1.x

File: common/config/azure-pipelines/templates/docs-build.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 parameters:
   - name: workingDir
     type: string
-    default: $(System.DefaultWorkingDirectory)
+    default: $(Build.SourcesDirectory)/itwinjs-core
```
Comment (calebmshafer) [inline common/config/azure-pipelines/templates/docs-build.yaml:7]: why is this needed? Are we checking out two repos?

File: common/config/azure-pipelines/templates/docs-build.yaml [hunk 2]
```diff
@@ -16,6 +16,12 @@ parameters:
     default: false
 
 steps:
+  - task: CopyFiles@2
+    displayName: Copy Files to $(Build.SourcesDirectory)\itwinjs-core
+    inputs:
+      SourceFolder: $(Build.SourcesDirectory)
+      TargetFolder: ${{ parameters.workingDir }}
```
Comment (calebmshafer) [inline common/config/azure-pipelines/templates/docs-build.yaml:23]: if this is the goal we can change the checkout step to directly clone it there.

File: common/config/azure-pipelines/jobs/docs-ci.yaml [hunk 3]
```diff
@@ -43,8 +44,11 @@ jobs:
...
       - template: ../templates/docs-build.yaml
+        parameters:
+          workingDir: $(Pipeline.Workspace)/itwinjs-core
+          outputDir: $(Agent.BuildDirectory)/tempDocsBuild/public_build
 
       - publish: $(Agent.BuildDirectory)/tempDocsBuild/
         artifact: DocsBuild
         displayName: Publish Pipeline Artifact - DocsBuild
-        condition: and(succeeded(), or(eq(variables['Build.SourceBranch'], 'refs/heads/master'), startsWith(variables['Build.SourceBranch'], 'refs/heads/release')))
+        condition: and(succeeded(), or(true, eq(variables['Build.SourceBranch'], 'refs/heads/master'), startsWith(variables['Build.SourceBranch'], 'refs/heads/release')))
```
Comment (skirby1996) [inline common/config/azure-pipelines/jobs/docs-ci.yaml:54]: This change is temporary while I test a deployment with the new changes, will be reverted.

File: common/config/azure-pipelines/templates/docs-build.yaml (hunk 1 above)
Comment (calebmshafer) [reply common/config/azure-pipelines/templates/docs-build.yaml]: sounds good.

**Review by calebmshafer — APPROVED** [review]
